lib/backend/k8s/resources/globalconfig.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/projectcalico/libcalico-go/lib/backend/k8s/thirdparty"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	GlobalConfigResourceName = "globalconfigs"
+	GlobalConfigTPRName      = "global-config.projectcalico.org"
+)
+
+func NewGlobalConfigClient(c *kubernetes.Clientset, r *rest.RESTClient) K8sResourceClient {
+	return &customK8sResourceClient{
+		clientSet:       c,
+		restClient:      r,
+		name:            GlobalConfigTPRName,
+		resource:        GlobalConfigResourceName,
+		description:     "Calico Global Configuration",
+		k8sResourceType: reflect.TypeOf(thirdparty.GlobalConfig{}),
+		k8sListType:     reflect.TypeOf(thirdparty.GlobalConfigList{}),
+		converter:       GlobalConfigConverter{},
+	}
+}
+
+// GlobalConfigConverter implements the K8sResourceConverter interface.
+type GlobalConfigConverter struct {
+}
+
+func (_ GlobalConfigConverter) ListInterfaceToKey(l model.ListInterface) model.Key {
+	pl := l.(model.GlobalConfigListOptions)
+	if pl.Name != "" {
+		return model.GlobalConfigKey{Name: pl.Name}
+	}
+	return nil
+}
+
+func (_ GlobalConfigConverter) KeyToName(k model.Key) (string, error) {
+	return strings.ToLower(k.(model.GlobalConfigKey).Name), nil
+}
+
+func (_ GlobalConfigConverter) NameToKey(name string) (model.Key, error) {
```

Comment:
So, because we lose the capitalization information in this method I think we probably don't want / need to implement this?


Basically, the lack of symmetry: `IpInIpEnabled -> ipinipenabled` != `ipipenabled -> ipipenabled`

It doesn't look like it's used in this file, and the only other place I see is here: https://github.com/projectcalico/libcalico-go/blob/cf00825e59c15c61741d13bbe10e5491997daf60/lib/backend/k8s/syncer.go#L1057

Might it be better to have this method return an error? I think the way it is might cause unexpected behavior since